packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -1125,30 +1157,24 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
       crossAxisUnconstrained: false,
       builder: (BuildContext context, MenuController controller, Widget? child) {
         assert(_initialMenu != null);
-        final bool isCollapsed = widget.inputDecorationTheme?.isCollapsed ?? false;
-        final Widget trailingButton = widget.showTrailingIcon
-            ? Padding(
-                padding: isCollapsed ? EdgeInsets.zero : const EdgeInsets.all(4.0),
-                child: IconButton(
-                  focusNode: _trailingIconButtonFocusNode,
-                  isSelected: controller.isOpen,
-                  constraints: widget.inputDecorationTheme?.suffixIconConstraints,
-                  padding: isCollapsed ? EdgeInsets.zero : null,
-                  icon: widget.trailingIcon ?? const Icon(Icons.arrow_drop_down),
-                  selectedIcon: widget.selectedTrailingIcon ?? const Icon(Icons.arrow_drop_up),
-                  onPressed: !widget.enabled
-                      ? null
-                      : () {
-                          handlePressed(controller);
-                        },
-                ),
-              )
-            : const SizedBox.shrink();
-
-        final Widget leadingButton = Padding(
-          padding: const EdgeInsets.all(8.0),
-          child: widget.leadingIcon ?? const SizedBox.shrink(),
+        final DropdownMenuDecorationBuilder decorationBuilder =
+            widget.decorationBuilder ?? _buildLegacyDecoration;
+        InputDecoration decoration = decorationBuilder(context, controller);
+        // If no suffixIcon is provided, the legacy IconButton is used for convenience.
+        if (decoration.suffixIcon == null) {
+          decoration = decoration.copyWith(suffixIcon: _buildLegacySuffixIcon(context, controller));
+        }
+        final InputDecoration effectiveDecoration = decoration.applyDefaults(
+          effectiveInputDecorationTheme,
         );
+        final InputDecoration keyedDecoration = effectiveDecoration.prefixIcon == null
```

Comment:
Can we just make changes on the `effectiveDecoration`? Feels a bit confused for the name of this variable😄

Reply:
> Can we just make changes on the effectiveDecoration?

Thats not possible because both fields are needed below.
The 'effectivedDecoration' is used in DropdownMenuBody and 'keyedDecoration' is used is passed to the TextField.
Maybe I can use 'textFieldDecoration' instead of 'keyedDecoration'? ("Keyed" was because the purpose of this decoration copy is to add a key to the prefixIcon, key which is used later for measurement).
I can also get rid of of the local variable and move the logic to line 1210 where this variable is used.